### PR TITLE
Updates UrlMap, BackendService, Healthcheck, Certs, InstanceGroup to use the generated code

### PR DESCRIPTION
**Pull request by bowei**
Updates UrlMap, BackendService, Healthcheck, Certs, InstanceGroup to use the generated code

```release-note
NONE
```

### CODE REVIEW

**Comment by bowei** [comment]
/assign @nicksardo @freehan 

depends on https://github.com/kubernetes/kubernetes/pull/58435

File: pkg/cloudprovider/providers/gce/gce_cert.go [hunk 1]
```diff
@@ -29,46 +33,33 @@ func newCertMetricContext(request string) *metricContext {
...
-
-	if err != nil {
-		if isHTTPErrorCode(err, http.StatusNotFound) {
-			return nil
-		}
-
-		return mc.Observe(err)
+	err := gce.c.SslCertificates().Delete(context.Background(), meta.GlobalKey(name))
+	if isHTTPErrorCode(err, http.StatusNotFound) {
+		err = nil
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_cert.go]: Don't think we need to return nil for NotFound anymore. Both ingress-gce and the e2e/ingress-utils check for NotFound.

File: pkg/cloudprovider/providers/gce/gce_backendservice.go [hunk 2]
```diff
@@ -34,150 +38,113 @@ func newBackendServiceMetricContextWithVersion(request, region, version string)
...
 func (gce *GCECloud) DeleteRegionBackendService(name, region string) error {
 	mc := newBackendServiceMetricContext("delete", region)
-	op, err := gce.service.RegionBackendServices.Delete(gce.projectID, region, name).Do()
-	if err != nil {
-		if isHTTPErrorCode(err, http.StatusNotFound) {
-			return nil
-		}
-		return mc.Observe(err)
+	err := gce.c.RegionBackendServices().Delete(context.Background(), meta.RegionalKey(name, region))
+	if isHTTPErrorCode(err, http.StatusNotFound) {
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_backendservice.go]: Believe we can remove this condition.  ILB checks for NotFound.

File: pkg/cloudprovider/providers/gce/gce_backendservice.go [hunk 3]
```diff
@@ -34,150 +38,113 @@ func newBackendServiceMetricContextWithVersion(request, region, version string)
...
 	mc := newBackendServiceMetricContext("delete", "")
-	op, err := gce.service.BackendServices.Delete(gce.projectID, name).Do()
-	if err != nil {
-		if isHTTPErrorCode(err, http.StatusNotFound) {
-			return nil
-		}
-		return mc.Observe(err)
+	err := gce.c.BackendServices().Delete(context.Background(), meta.GlobalKey(name))
+	if isHTTPErrorCode(err, http.StatusNotFound) {
+		return nil
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_backendservice.go]: Can cut. Both e2e/ingress-utils and ingress-gce check for NotFound.

File: pkg/cloudprovider/providers/gce/gce_targetproxy.go [hunk 4]
```diff
@@ -29,49 +33,37 @@ func newTargetProxyMetricContext(request string) *metricContext {
...
 	mc := newTargetProxyMetricContext("delete")
-	op, err := gce.service.TargetHttpProxies.Delete(gce.projectID, name).Do()
-	if err != nil {
-		if isHTTPErrorCode(err, http.StatusNotFound) {
-			return nil
-		}
-		return mc.Observe(err)
+	err := gce.c.TargetHttpProxies().Delete(context.Background(), meta.GlobalKey(name))
+	if isHTTPErrorCode(err, http.StatusNotFound) {
+		err = nil
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_targetproxy.go]: Remove condition. Handled in ingress-gce

File: pkg/cloudprovider/providers/gce/gce_targetproxy.go [hunk 5]
```diff
@@ -80,59 +72,45 @@ func (gce *GCECloud) ListTargetHttpProxies() (*compute.TargetHttpProxyList, erro
...
 	mc := newTargetProxyMetricContext("delete")
-	op, err := gce.service.TargetHttpsProxies.Delete(gce.projectID, name).Do()
-	if err != nil {
-		if isHTTPErrorCode(err, http.StatusNotFound) {
-			return nil
-		}
-		return mc.Observe(err)
+	err := gce.c.TargetHttpsProxies().Delete(context.Background(), meta.GlobalKey(name))
+	if isHTTPErrorCode(err, http.StatusNotFound) {
+		err = nil
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_targetproxy.go]: Remove condition. Handled in ingress-gce

File: pkg/cloudprovider/providers/gce/gce_urlmap.go [hunk 6]
```diff
@@ -29,47 +33,35 @@ func newUrlMapMetricContext(request string) *metricContext {
...
 	mc := newUrlMapMetricContext("delete")
-	op, err := gce.service.UrlMaps.Delete(gce.projectID, name).Do()
-	if err != nil {
-		if isHTTPErrorCode(err, http.StatusNotFound) {
-			return nil
-		}
-		return mc.Observe(err)
+	err := gce.c.UrlMaps().Delete(context.Background(), meta.GlobalKey(name))
+	if isHTTPErrorCode(err, http.StatusNotFound) {
+		err = nil
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_urlmap.go]: Remove condition. Both test/ingress-utils and ingress-gce check for NotFound.

File: pkg/cloudprovider/providers/gce/gce_cert.go (hunk 1 above)
Comment (bowei) [reply pkg/cloudprovider/providers/gce/gce_cert.go]: done

File: pkg/cloudprovider/providers/gce/gce_backendservice.go (hunk 2 above)
Comment (bowei) [reply pkg/cloudprovider/providers/gce/gce_backendservice.go]: done

File: pkg/cloudprovider/providers/gce/gce_backendservice.go (hunk 3 above)
Comment (bowei) [reply pkg/cloudprovider/providers/gce/gce_backendservice.go]: done

File: pkg/cloudprovider/providers/gce/gce_targetproxy.go (hunk 4 above)
Comment (bowei) [reply pkg/cloudprovider/providers/gce/gce_targetproxy.go]: done

File: pkg/cloudprovider/providers/gce/gce_targetproxy.go (hunk 5 above)
Comment (bowei) [reply pkg/cloudprovider/providers/gce/gce_targetproxy.go]: done

File: pkg/cloudprovider/providers/gce/gce_urlmap.go (hunk 6 above)
Comment (bowei) [reply pkg/cloudprovider/providers/gce/gce_urlmap.go]: done

**Pull request merged**